### PR TITLE
Sane dump

### PR DESCRIPTION
**Pull request by RenaudWasTaken**
Ok so this pull request should not be merged / rebased at this point
This is just so that I get some feedback on everything.

Things which are done in this :
- Builds
- Passes Tests
- Implements new Protocol
- Adds a lot of Debug Prints
- Mostly seems to work on a regular game

Things which I'd like advice:
- Does this code seem crazy (except the fact that there aren't any observers) ?
- Should I keep the Debugs ?
- Do you see any bugs ?

Things which will be done:
- Add more tests
- Reformat commit history

Thanks !

### CODE REVIEW

File: src/rules.cc [hunk 1]
```diff
@@ -100,19 +128,23 @@ void Rules::end_of_round() {
...
+  // Only called on player who played
+  if (game == MUR && api_->game_state()->get_mur_used_stock(player) == -1)
+    handle_timeout(player);
+  else if (game == NOSE &&
+           api_->game_state()->get_nose_played_square(player).x == -1)
+    handle_timeout(player);
 }
 
 void Rules::dump_state(std::ostream& out) {
+  DEBUG("[SERVER] Dumping state!");
```
Comment (adrientoub) [inline src/rules.cc]: SDEBUG?

File: src/api.cc [hunk 2]
```diff
@@ -33,6 +33,7 @@ action_error Api::play_mur(mur_position position, int amount) {
 
   actions_.add(action);
   game_state_set(action->apply(game_state_));
+  DEBUG("[ME-%d] used stock: %d", me(), game_state()->get_mur_used_stock(me()));
```
Comment (adrientoub) [inline src/api.cc]: PDEBUG?

Comment (RenaudWasTaken) [inline src/api.cc]: Won't work on this one :D

File: src/rules.cc (hunk 1 above)
Comment (RenaudWasTaken) [inline src/rules.cc]: I'll fix that :D

**Pull request merged**